apiserver/facades/agent/caasagent/caasagent.go

```diff
@@ -9,21 +9,22 @@ import (
 	"github.com/juju/juju/apiserver/common"
 	"github.com/juju/juju/apiserver/common/cloudspec"
 	"github.com/juju/juju/apiserver/facade"
-	"github.com/juju/juju/apiserver/params"
 )
 
 type Facade struct {
 	auth      facade.Authorizer
 	resources facade.Resources
-	state     CAASAgentState
 	cloudspec.CloudSpecAPI
-
-	model Model
+	*common.ModelWatcher
 }
 
 // NewStateFacade provides the signature required for facade registration.
 func NewStateFacade(ctx facade.Context) (*Facade, error) {
 	authorizer := ctx.Auth()
+	if !authorizer.AuthMachineAgent() {
```

Comment:
When would this trigger, or is this just a catch all?

Reply:
It's a catch all - our standard way of excluding other agent types